src/modules/sync/syncModelRun.ts

```diff
@@ -75,11 +130,14 @@ export const getDiffModel = async (
     params,
   );
 
+  const filteredSourceModel = filterModel(sourceModel, params.entities);
+  const filteredTargetModel = filterModel(transformedTargetModel, params.entities);
```

Comment:
It'd be better not to even fetch the entities that are not supposed to be synced. To avoid unnecessary fetch of all the entities when one wants to sync just collections for example.

Reply:
how would this work? we usually fetch all units from an entity, as we do not have filtering option in Management SDK. What I can do is that those undefined properties in entities option I can prevent those from fetching.. However still need to filter here. What is more some might be still needed for transformation (ids -> to codenames), for example types needs snippets and so on

Reply:
I meant to not fetch the entities at all when they are not included. Of course, we cannot filter them. To make sure the transformations work, we can maintain a map of dependencies between the entities and always fetch all the dependencies of all the included entities. That shouldn't be very complicated.